doozer/doozerlib/distgit.py

```diff
@@ -1672,48 +1679,31 @@ def _determine_upstream_rhel_version(self, source_path) -> Optional[int]:
 
         # We will infer the rhel version from the last build layer in the upstream Dockerfile
         last_layer_pullspec = parent_images[-1]
-        redis_key = f'appdata:upstream-el-version:{last_layer_pullspec}'
-
-        # If the pullspec is not tracked in Redis yet, this will return None
-        version = redis.get_value_sync(redis_key)
-        if version:
-            # Already tracked in Redis
-            return int(version)
 
-        # Not tracked yet: let's read it from os-release
+        version = None
         try:
-            # Create a temp dir with a random name to store the os-release file
-            with TemporaryDirectory() as tmpdir:
-                # Extract os-release
-                cmd = ['oc', 'image', 'extract', f'--path=/usr/lib/os-release:{tmpdir}',
-                       last_layer_pullspec, '--confirm']
-                exectools.cmd_assert(cmd)
-
-                # Parse it
-                with open(f'{tmpdir}/os-release') as os_release:
-                    # os-release will contain a line like this: VERSION_ID="9.2"
-                    # We need to eval the result to turn '"9.2"' into '9.2'
-                    version_id = eval([line for line in os_release if 'VERSION_ID' in line][0].split('=')[1])
-                    version = isolate_rhel_major_from_version(version_id)
-                    self.logger.info('Intended upstream RHEL version for %s: %s', self.name, version)
+            # We will infer the rhel version from the last build layer in the upstream Dockerfile
+            last_layer = dfp.parent_images[-1]
```

Comment:
Already done at line [1681](https://github.com/openshift-eng/art-tools/pull/361/files#diff-d6994e859542b479607bede445901c422a82513def8423a189876ef9ca601b6cL1674)

Reply:
Good catch. Late night merge conflict resolution should be avoided :) .